core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -1534,10 +1533,12 @@ class Partition(val topicPartition: TopicPartition,
       partitionEpoch
     )
     val updatedState = PendingExpandIsr(
-      partitionState.isr,
       newInSyncReplicaId,
       newLeaderAndIsr,
-      partitionState
+      // The current partition state must be of type CommittedPartitionState
+      // if we are here. CommittedPartitionState is the only one with `isInflight`
+      // equals to false.
+      partitionState.asInstanceOf[CommittedPartitionState]
```

Comment:
Perhaps a nicer way we can do this is with a `match` in the caller. Something like this:
```scala
  partitionState match {
    case currentState: CommittedState if needsExpandIsr(followerReplica) => 
      Some(prepareIsrExpand(currentState, followerReplica.brokerId))
    case _ =>
      None 
  }
```